Detectors/ITSMFT/common/reconstruction/src/LookUp.cxx

```diff
@@ -29,17 +29,41 @@ LookUp::LookUp(std::string fileName)
   mTopologiesOverThreshold = mDictionary.mFinalMap.size();
 }
 
-int LookUp::findGroupID(int nRow, int nCol, const unsigned char patt[Cluster::kMaxPatternBytes], int nBytesUsed)
+int LookUp::groupFinder(int nRow, int nCol)
 {
-  unsigned long hash = ClusterTopology::getCompleteHash(nRow, nCol, patt, nBytesUsed);
+  int row_index = nRow / TopologyDictionary::RowClassSpan;
+  int col_index = nCol / TopologyDictionary::ColClassSpan;
+  if (row_index > TopologyDictionary::MaxNumberOfClasses || col_index > TopologyDictionary::MaxNumberOfClasses) {
+    return TopologyDictionary::NumberOfRareGroups - 1;
+  } else {
+    return row_index * TopologyDictionary::MaxNumberOfClasses + col_index;
+  }
+}
+
+int LookUp::findGroupID(int nRow, int nCol, const unsigned char patt[Cluster::kMaxPatternBytes])
+{
+  int nBits = nRow * nCol;
+  int nBytes = nBits / 8;
```

Comment:
You don't need actually to spend time on nBytes calculation, can check directly ````if (nBits<9) { // small topology````

Reply:
You're right: done